qualtran/bloqs/arithmetic/multiplication_test.py

```diff
@@ -119,6 +120,94 @@ def test_product():
     num_toff = 2 * bitsize * mbits - max(bitsize, mbits)
     assert t_complexity(cbloq) == TComplexity(t=4 * num_toff)
 
+def test_product_gates():
```

Comment:
we have methods that do most of the testing, you can use these as example

https://github.com/quantumlib/Qualtran/blob/60037bcb9e088d0e22ef436360e6ec262f4cfe88/qualtran/bloqs/mod_arithmetic/mod_subtraction_test.py#L209-L248